docs/resources/kafka_topic.md

```diff
@@ -0,0 +1,151 @@
+```
+
+### Import a topic and its configs
+```shell
+terraform import instaclustr_kafka_topic.<resource-name> "<cluster-id>&<topic-name>&config"
+```
+Replace `<resource-name>`, `<cluster-id>`, and `<topic-name>` with the real ones, keep this "&" symbol in the command.
+
+### Example
+```shell
+terraform import instaclustr_kafka_topic.kafka_topic_test "f4cb7a63-8217-4dc8-ab12-28e54efc00d1&test"
+terraform import instaclustr_kafka_topic.kafka_topic_test "f4cb7a63-8217-4dc8-ab12-28e54efc00d1&test&config"
+```
+
+After importing, you should see the imported topic in ***terraform.tfstate***.
+
+## Schema
+Property | Description | Default
+---------|-------------|--------
+cluster_id | The ID of an existing Instaclustr Kafka managed cluster. | Required
+topic | The topic name for the Kafka topic. | Required
+replication_factor | The replication factor of this topic. | Required
+partitions | The partitions number of this topic. | Required
+config | The list of this topic's configs, see below for its properties. | Optional, but required if you want to update the topic's configs
+
+### config
+This contains all the possible configs that you can change for a Kafka topic. We use Kafka AdminClient 2.3.1 internally, please see https://kafka.apache.org/23/documentation.html#topicconfigs
+to check every config's information. 
+
+Property | Data Type | Default
+---------|-------------|--------
+compression_type | string | None
+leader_replication_throttled_replicas | string | ""
+min_insync_replicas | int | None
+message_downconversion_enable | boolean | true
+segment_jitter_ms | int | 0
+cleanup_policy | string | None
+flush_ms | string<sup>* | None
+follower_replication_throttled_replicas | string | None
+retention_ms | int | None
+segment_bytes | int | None
+flush_messages | string<sup>* | None
+message_format_version| string | None
+file_delete_delay_ms | int | None
+max_compaction_lag_ms | string<sup>* | None
+max_message_bytes | int | None
+min_compaction_lag_ms | int | 0
+message_timestamp_type | string | None
+preallocate | boolean | false
+index_interval_bytes | int | None
+min_cleanable_dirty_ratio | float | None
+unclean_leader_election_enable | boolean | false
+delete_retention_ms | int | None
+retention_bytes | int | None
+segment_ms | int | None
+message_timestamp_difference_max_ms | string<sup>* | None
+segment_index_bytes | int | None
+
+<sup>* Indeed an int value, using string because Terraform has difficulties parsing big int, it will lose precision.
+
+## Usage
+There are two use cases for updating a Kafka topic's configs.
+1. Providing all the possible configs in resource, and modify the ones you want to update. Then keeping all the configs in the resource.
+2. Providing some configs (not all) you want to update, and then keep only these configs in the resource.
+
+Terraform always detects that there is an update in case 2 if you run the plan again. Although Terraform detects an update, we make sure
```

Comment:
This still isn't great. Terraform says one thing, but we do something else. We might need to re-discuss the option to suppress the diff if the property is null.

Reply:
I don't think there is an easy way to handle this. By saying this in doc, we are actually encouraging our customers to use the first way.  They can store all the configs in the resource and change the ones they want to change. The first way is also better in terms of keeping a record of all the topic configs for latter usages and references.